panic in fsmonitor

https://jenkins.linuxcontainers.org/job/lxd-github-pull-test/5100/arch=amd64,backend=dir,compiler=golang-tip/console#console-section-2

INFO[11-03|19:05:50] Loading daemon configuration 
INFO[11-03|19:05:50] Initializing networks 
INFO[11-03|19:05:50] Pruning leftover image files 
INFO[11-03|19:05:50] Done pruning leftover image files 
INFO[11-03|19:05:50] Starting device monitor 
WARN[11-03|19:05:50] Failed to initialize fanotify, falling back on fsnotify err="Failed to watch directory \"/dev\": no such device"
INFO[11-03|19:05:50] Daemon stopped 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xe24ed6]

goroutine 1 [running]:
github.com/lxc/lxd/lxd/fsmonitor/drivers.(*fsnotify).watchFSTree.func1({0xc000201f80, 0x1a}, {0x0, 0x0}, {0x1bad020?, 0xc0005a7770})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/drivers/driver_fsnotify.go:141 +0x56
path/filepath.walk({0x1897ad2, 0x4}, {0x1bbe598, 0xc0000fcea0?}, 0xc000849000)
	/snap/go_tip/8614/src/path/filepath/path.go:438 +0x20d
path/filepath.Walk({0x1897ad2, 0x4}, 0xc000849000)
	/snap/go_tip/8614/src/path/filepath/path.go:505 +0x6c
github.com/lxc/lxd/lxd/fsmonitor/drivers.(*fsnotify).watchFSTree(0xc0002a4cc0, {0x1897ad2, 0x4})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/drivers/driver_fsnotify.go:139 +0x6a
github.com/lxc/lxd/lxd/fsmonitor/drivers.(*fsnotify).load(0xc0002a4cc0, {0x1bb89a0?, 0xc000576480})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/drivers/driver_fsnotify.go:36 +0xc5
github.com/lxc/lxd/lxd/fsmonitor/drivers.Load({0x1bb89a0, 0xc000576480}, {0x7f635800e968, 0xc00049ca00}, {0x18a54de?, 0x18974dc?}, {0x1897ad2, 0x4})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/drivers/load.go:25 +0xb8
github.com/lxc/lxd/lxd/fsmonitor.New.func1({0x18a54de, 0x8})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/load.go:19 +0x118
github.com/lxc/lxd/lxd/fsmonitor.New({0x1bb89a0?, 0xc000576480?}, {0x1897ad2, 0x4})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/fsmonitor/load.go:34 +0x1ce
main.(*Daemon).init(0xc0005243c0)
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/daemon.go:1347 +0x3813
main.(*Daemon).Init(0xc0005243c0)
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/daemon.go:740 +0x5f
main.(*cmdDaemon).Run(0xc000556b88, 0xc000575c40?, {0xc0005590e0, 0x0, 0x0?})
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/main_daemon.go:78 +0x54f
github.com/spf13/cobra.(*Command).execute(0xc00044b680, {0xc0001b8010, 0x3, 0x3})
	/lxc-ci/build/tmp.Pqleolkg34/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00044b680)
	/lxc-ci/build/tmp.Pqleolkg34/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/lxc-ci/build/tmp.Pqleolkg34/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
	/lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/lxd/main.go:222 +0x1a55
Error: LXD still not running after 300s timeout (Get "http://unix.socket/1.0": dial unix /lxc-ci/build/tmp.Pqleolkg34/go/src/github.com/lxc/lxd/test/tmp.jx7/IDUGS7Dbn/unix.socket: connect: connection refused)
Killing PID 1466034
./main.sh: 50: kill: No such process

--- a/lxd/fsmonitor/drivers/driver_fsnotify.go
+++ b/lxd/fsmonitor/drivers/driver_fsnotify.go
@@ -137,18 +137,18 @@ func (d *fsnotify) watchFSTree(path string) error {
 	}
 
 	err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
-		// Ignore files and symlinks
-		if !info.IsDir() || info.Mode()&os.ModeSymlink != 0 {
-			return nil
-		}
-
-		// Check for errors here as we only care about directories. Files and symlinks are aren't of interest for this.
+		// Check for errors here as we only care about directories. Files and symlinks aren't of interest for this.
 		if err != nil {
 			d.logger.Warn("Error visiting path", log.Ctx{"path": path, "err": err})
 			return nil
 		}
 
-		// Only watch on real paths
+		// Ignore files and symlinks.
+		if !info.IsDir() || info.Mode()&os.ModeSymlink != 0 {
+			return nil
+		}
+
+		// Only watch on real paths.
 		err = d.watcher.Add(path)
 		if err != nil {
 			d.logger.Warn("Failed to watch path", log.Ctx{"path": path, "err": err})

